### PR TITLE
resource_control: fix burst patch

### LINKED ISSUE #48140: Resource group's burstlimit is wrong after ALTER.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
create resource group rg3 ru_per_sec=5000;
alter resource group rg3 ru_per_sec=2000;
```

### 2. What did you expect to see? (Required)
burst_limit = 2000
### 3. What did you see instead (Required)
image

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: close #48140

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by CabinfeverB** [comment]
PTAL @nolouch @glorv

**Review by glorv — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
ptal @zimulala

File: pkg/ddl/tests/resourcegroup/resource_group_test.go [hunk 1]
```diff
@@ -137,6 +137,16 @@ func TestResourceGroupBasic(t *testing.T) {
 	}
 	g = testResourceGroupNameFromIS(t, tk.Session(), "y")
 	checkFunc(g)
+	tk.MustExec("alter resource group y RU_PER_SEC=6000")
```
Comment (nolouch) [inline pkg/ddl/tests/resourcegroup/resource_group_test.go:140]: can u add test about `alter resource group y  PRIORITY=hight`?

File: pkg/ddl/tests/resourcegroup/resource_group_test.go [hunk 2]
```diff
@@ -137,6 +137,16 @@ func TestResourceGroupBasic(t *testing.T) {
...
+	checkFunc = func(groupInfo *model.ResourceGroupInfo) {
+		re.Equal(true, groupInfo.ID != 0)
+		re.Equal("y", groupInfo.Name.L)
+		re.Equal(groupID.Load(), groupInfo.ID)
+		re.Equal(uint64(6000), groupInfo.RURate)
+		re.Equal(int64(6000), groupInfo.BurstLimit)
```
Comment (zimulala) [inline pkg/ddl/tests/resourcegroup/resource_group_test.go:146]: In the issue， we can see the result as follows：
```
create resource group rg3 ru_per_sec=2000;
alter resource group rg3 ru_per_sec=5000;
2. What did you expect to see? (Required)
burst_limit = 2000
```
So why here is `6000` not `4000`?

Comment (CabinfeverB) [reply pkg/ddl/tests/resourcegroup/resource_group_test.go:146]: Sry, I wrote the wrong ddl statement in the issue. It should be 
`create resource group rg3 ru_per_sec= 5000; `
`alter resource group rg3 ru_per_sec=2000;`

**Review by nolouch — APPROVED** [review]
lgtm

**Review by zimulala — APPROVED** [review]

**Pull request merged**